internal/provider/provider.go

```diff
@@ -320,30 +322,64 @@ func azureProvider(supportLegacyTestSuite bool) *schema.Provider {
 			"features": schemaFeatures(supportLegacyTestSuite),
 
 			// Advanced feature flags
+			"resource_provider_registrations": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				DefaultFunc: schema.EnvDefaultFunc("ARM_RESOURCE_PROVIDER_REGISTRATIONS", "legacy"),
+				Description: "The set of Resource Providers which should be automatically registered for the subscription.",
+				ValidateFunc: validation.StringInSlice([]string{
+					"core",
+					"extended",
+					"all",
+					"none",
```

Comment:
I'm unsure about the naming of the mode `none` here, specifically when used in conjunction with `resource_providers_to_register` like below, for `none` I would really expect no RPs to be registered at all, even user specified ones
```
provider "azurerm" {
    resource_provider_registrations = "none"

    resource_providers_to_register = [
        "Microsoft.ApiManagement",
        "Microsoft.Compute",
        "Microsoft.KeyVault",
        "Microsoft.Network",
        "Microsoft.Storage",
    ]
}
```

Tom made the suggestion of naming it `manual`, my suggestion would have been `custom`

Reply:
In this implementation, `resource_providers_to_register` works in addition to `resource_provider_registrations`, so I'm not sure that `manual` or `custom` confer the expected behavior? Perhaps the property names also need improvement.

Reply:
maybe if the property name changes? `resource_provider_registrations` to `resource_provider_group_to_register`? grouping? collection? or something else then none would make sense.